kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -296,16 +308,11 @@ public synchronized void startAllRegisteredInformers() {
     }
 
     if (!informerExecutor.isShutdown()) {
-      informers.forEach(
-          (informerType, informer) -> startedInformers.computeIfAbsent(informerType,
-              key -> informerExecutor.submit(() -> {
-                try {
-                  informer.run();
-                } catch (RuntimeException e) {
-                  this.eventListeners.forEach(listener -> listener.onException(informer, e));
-                  log.warn("Informer start did not complete", e);
-                }
-              })));
+      List<Map.Entry<SharedIndexInformer, Future<Boolean>>> startInformerTasks = new ArrayList<>();
+      for (Map.Entry<OperationContext, SharedIndexInformer> entry : informers) {
+        startInformerTasks.add(new AbstractMap.SimpleEntry<>(entry.getValue(), startInformerAsync(entry.getValue())));
+      }
+      waitForAllInformersToStart(startInformerTasks);
```

Comment:
waiting doesn't match previous behavior, the other java client, nor the go client.  Perhaps returning a future would be the best compromise.  It would be easy to create a CompleteableFuture of all the starts.